src/dotnet/MsbuildProject.cs

```diff
@@ -90,22 +103,7 @@ public static MsbuildProject FromDirectory(ProjectCollection projects, string pr
                 throw new GracefulException(CommonLocalizableStrings.MoreThanOneProjectInDirectory, projectDirectory);
             }
 
-            FileInfo projectFile = files.First();
-
-            if (!projectFile.Exists)
-            {
-                throw new GracefulException(
-                    CommonLocalizableStrings.CouldNotFindAnyProjectInDirectory,
-                    projectDirectory);
-            }
-
-            var project = TryOpenProject(projects, projectFile.FullName);
-            if (project == null)
-            {
-                throw new GracefulException(CommonLocalizableStrings.FoundInvalidProject, projectFile.FullName);
-            }
-
-            return new MsbuildProject(projects, project);
+            return files.First();
```

Comment:
This was written by @krwq . But, since we throw if files.Count ==0  or > 1, this should be fine.

https://github.com/mishra14/cli/blob/ac988a4ac92cecb3eb21a58e88cf645f0f2d7279/src/dotnet/MsbuildProject.cs#L94